File: JavaScript/0151_reverse_words_in_string/solution.js
function reverse_words(string) {
    string = string.split(" ");
    let word_list = new Array();
    for (let word of string) {
        if (word !== "") {
            word_list.push(word);
        }
    }
    let result = new String();
    for (let index = word_list.length -1; index >= 0; index --) {
        if (index === 0) {
            result = result + word_list[index];
        } else {
            result = result + word_list[index];
            result = result + " ";
        }
    }
    return result;
}

let string = " hello world ";
console.log(reverse_words(string));

File: Python/0151_reverse_words_in_string/solution.py
def reverse_words(string):
    string = string.split()
    result = str()
    for index in range(len(string) -1, -1, -1):
        if index == 0:
            result = result + string[index]
        else:
            result = result + string[index]
            result = result + " "
    return result

string = "  the sky is blue  "
print(reverse_words(string))